clean architecture and final features

Vaniahchristian opened the pull request:
**what does this PR do** 
 It implements the clean architecture and the final features of the application

tasks to be completed
-clean architecture
-Add the open device settings 
-filter by name , number of repositories, number of followers
-Get rid of the card on my bio data
-Fix my data  connectivity function
-Add label on my filter bottom sheet

Vaniahchristian committed:
clean architecture and final features

Vaniahchristian committed:
Update user details screen: Updated user_details.dart with new data fetching logic to enable profile opening by tapping the link.

NabunyaLilian commented on lib/Data/Remote_Data_Source/github_service.dart:
Instead of returning FutureList<dynamic> consider returning the actual defined objects.

Vaniahchristian committed:
Clean architecture

Made changes to my filter bottom sheet

Made changes to my user details: removed the card from bio

fixed my data connectivity function

Vaniahchristian committed:
MY USER PROVIDER IMPLEMENTATION

Vaniahchristian committed:
edit on the future builder

Vaniahchristian committed:
code review

Vaniahchristian committed:
changed the dynamic in the remote datasource to GithubUserModel

Vaniahchristian committed:
User Details Provider

Vaniahchristian committed:
Finished working on the UI and creating files for the widgets

Vaniahchristian committed:
error to be handled in the repository implementation
